tests/acceptance/features/apiSpaces/listSpaces.feature

```diff
@@ -19,6 +19,35 @@ Feature: List and create spaces
       | name             | Alice Hansen |
       | quota@@@state    | normal       |
       | root@@@webDavUrl | %base_url%/dav/spaces/%space_id% |
+    And the json responded should contain a space "Shares Jail" with these key and value pairs:
+      | key              | value        |
+      | driveType        | virtual     |
+      | id               | %space_id%   |
+      | name             | Shares Jail |
+      | root@@@webDavUrl | %base_url%/dav/spaces/%space_id% |
+
+  Scenario: An ordinary user can request information about their Space via the Graph API using a filter
+    When user "Alice" lists all available spaces via the GraphApi with query "$filter=driveType eq 'personal'"
+    Then the HTTP status code should be "200"
+    And the json responded should contain a space "Alice Hansen" with these key and value pairs:
+      | key              | value        |
+      | driveType        | personal     |
+      | id               | %space_id%   |
+      | name             | Alice Hansen |
+      | quota@@@state    | normal       |
+      | root@@@webDavUrl | %base_url%/dav/spaces/%space_id% |
+    And the json responded should not contain a space with name "Shares Jail"
+    And the json responded should contain spaces of type "personal"
+    And the json responded should only contain spaces of type "personal"
+    And the json responded should not contain spaces of type "project"
+    And the json responded should not contain spaces of type "virtual"
+    And the json responded should not contain spaces of type "public"
```

Comment:
not the same. "only" doesn't allow other space types while "contains"

Reply:
yes, maybe. This is a demo case ;-)

But in the future you may have two combined filters where `only` will not work.

Reply:
let me add another one to remove redundancy here and use the `not` logic in the other case. These then steps are not expensive because they check the existing response from the last spaces listing.